core/src/test/java/org/elasticsearch/index/query/RegexpQueryBuilderTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.RegexpQuery;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+
+public class RegexpQueryBuilderTest extends BaseQueryTestCase<RegexpQueryBuilder> {
+
+    private static final String[] REWRITE_METHOD = new String[]{"constant_score_auto", "scoring_boolean", "constant_score_boolean", 
+            "constant_score_filter", "top_terms_boost_1", "top_terms_1"};
+    
+    @Override
+    protected RegexpQueryBuilder doCreateTestQueryBuilder() {
+        RegexpQueryBuilder query = newRandomQueryBuilder();
+
+        if (randomBoolean()) {
+            List<RegexpFlag> flags = new ArrayList<>(); // used to test combination of flags
+            for (RegexpFlag flag : RegexpFlag.values()) {
```

Comment:
why not deciding upfront how many flags you want (randomized) and then call randomFrom(RegexpFlag.values()) n times?